core-processor/src/ext.rs

```diff
@@ -1051,12 +1092,14 @@ impl Externalities for Ext {
     }
 
     fn lock_payload(&mut self, at: u32, len: u32) -> Result<PayloadSliceLock, Self::FallibleError> {
-        let end = at
-            .checked_add(len)
-            .ok_or(FallibleExecutionError::TooBigReadLen)?;
-        self.charge_gas_runtime_if_enough(RuntimeCosts::ReadPerByte(len))?;
-        PayloadSliceLock::try_new((at, end), &mut self.context.message_context)
-            .ok_or_else(|| FallibleExecutionError::ReadWrongRange.into())
+        self.atomic_gas(|ext| {
+            let end = at
```

Comment:
fix diff pls here and above, for fns below GitHub shows real changes instead of just +-